File: Homework-by-C++/Homework-6/Hw 6_1.cpp
#include <iostream>

using namespace std;

void exchange(int arr[], int i, int j)
{
    int temp = arr[i];
    arr[i] = arr[j];
    arr[j] = temp;
}

void compare(int arr[], int i, int j, bool& exchanged)
{
    if(arr[i] > arr[j])
    {
        exchange(arr, i, j);
        exchanged = true;
    }
}

void display(int arr[], int N)
{
    for(int i = 0; i < N; i++)
    {
        if(i != N-1)
            cout << arr[i] << " ";
        else
            cout << arr[i];
    }
    cout << endl;
}

int main()
{
    int N;
    cin >> N;
    int arr[N];
    for(int i = 0; i < N; i++)
    {
        cin >> arr[i];
    }

    for(int i = 0; i < N-1; i++)
    {
        bool exchanged = false;
        for(int j = 0; j < N-1-i; j++)
        {
            compare(arr, j , j+1, exchanged);
        }
        if(!exchanged)
            break;
        else
            display(arr, N);
    }

    system("pause");
    return 0;
}

File: Homework-by-python/Homework-6/Hw 6_1.py
def exchange(arr, i, j):
    arr[i], arr[j] = arr[j], arr[i]

def compare(arr, i, j, exchanged):
    if arr[i] > arr[j]:
        exchange(arr, i, j)
        exchanged[0] = True  # Use a mutable type like a list to reflect changes

def display(arr):
    print(" ".join(map(str, arr)))

def main():
    N = int(input())
    arr = list(map(int, input().split()))

    for i in range(N-1):
        exchanged = [False]
        for j in range(N-1-i):
            compare(arr, j, j+1, exchanged)
        if not exchanged[0]:
            break
        display(arr)

if __name__ == "__main__":
    main()